website/content/docs/agent/index.mdx

```diff
@@ -8,18 +8,94 @@ description: |-
 
 # Vault Agent
 
-Vault Agent is a client daemon that provides the following features:
+Nearly all requests to Vault must be accompanied by a valid client token. This
+includes all API requests, as well as via the Vault CLI and other libraries.
+Therefore, Vault clients must first authenticate with Vault to acquire a token,
+and Vault provides a number of different authentication methods to assist in the
+delivery of this initial token.
+
+![Client authentication](/img/diagram-vault-agent.png)
+
+If the client can securely get the token, all subsequent requests (e.g. request
+database credentials, read key/value secrets) can be processed based on the
+trust established by successful authentication.
+
+This means that the client application must invoke the Vault API to authenticate
+with Vault and manage the acquired token in addition to invoking the API to
+request secrets from Vault. This implies code changes to client applications;
+therefore, additional testing and maintenance of the application.
+
+The following code example implements Vault API to authenticate with Vault
```

Comment:
It looks like this snippet is derived from the [documentation example](https://www.vaultproject.io/docs/auth/approle#code-example) so I did not try to build a version of it an run it.

Reply:
I think it's a great example, I was just saying I didn't test the part used here. 😀